### PR TITLE
fix: update installer to use https for download

### PR DESCRIPTION
**Pull request by eheyder**
fix error with 403 on http://s3-us-west-2.amazonaws.com/dynamodb-local/dynamodb_local_latest.tar.gz

### CODE REVIEW

**Review by sethalt — APPROVED** [review]
Thanks!

**Review by yanaSelin — APPROVED** [review]

**Review by jnorris-rbi — APPROVED** [review]
LGTM, thanks for tackling this change!

**Review by pablogdnd — APPROVED** [review]
LGTM

**Review by jphechter — APPROVED** [review]
Please add this.

**Comment by pablogdnd** [comment]
@AshanFernando can you check at this?

**Review by josephobiagba — APPROVED** [review]

**Comment by DeeWBee** [comment]
@AshanFernando several projects are now falling apart. Please fix this.

**Review by sachet-mittal — APPROVED** [review]

**Review by yaguchii — APPROVED** [review]
merge please

**Review by kooooichi24 — APPROVED** [review]

**Comment by kairi4** [comment]
@rehrumesh @mjzone 
Please review and merge this PR if possible

**Review by yoannbp — APPROVED** [review]

File: dynamodb/installer.js [hunk 1]
```diff
@@ -3,7 +3,7 @@
 var tar = require("tar"),
   zlib = require("zlib"),
   path = require("path"),
-  http = require("http"),
+  https = require("https"),
```
Comment (huseyin-caglayan-cko) [inline dynamodb/installer.js:6]:
```
    http = require('http'),
    https = require('https'),
```

File: dynamodb/installer.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ var download = function(downloadUrl, installPath, callback) {
   console.log(
     `Started downloading dynamodb-local from ${downloadUrl} into ${installPath}. Process may take few minutes.`
   );
-  http
+  https
```
Comment (huseyin-caglayan-cko) [inline dynamodb/installer.js:15]:
```
    const client = downloadUrl.toLowerCase().startsWith('https') ? https : http;
    client
        .get(downloadUrl, function (response) {

```

File: dynamodb/installer.js (hunk 1 above)
Comment (huseyin-caglayan-cko) [reply dynamodb/installer.js:6]: keeps support for http as well as https

**Comment by vestel** [comment]
Looks good! Please approve!

**Comment by LeoFalco** [comment]
merge please

**Review by willstodev — APPROVED** [review]
merge, please?

**Comment by DeeWBee** [comment]
Circling back here: I ditched this package (along with `severless-dynamodb-local`) and resorted to using [Docker + DynamoDB](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBLocal.DownloadingAndRunning.html#DynamoDBLocal.DownloadingAndRunning.title). I modified the `docker-compose` file to use my desired port **but otherwise kept the default configuration in place (as provided by AWS).** 

From there, I [manually created needed tables inside the Docker image](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/getting-started-step-1.html). To create tables in the Docker image, you append `--endpoint-url http://localhost:[YOUR_PORT_HERE]` to the regular `create table` command. From there, I was able to start my app using `sls offline start`. It's clear the author of these packages won't be updating anytime soon. I suggest people implement similar solutions and move off this expired package. 

I'll update this comment once the problematic packages have been completely removed from my app.

UPDATE: I removed the problematic packages from my app, and everything still works. I am officially done using these. Like I said, I suggest people do the same.

**Review by jlau-neonone — APPROVED** [review]
This fix worked for me.

**Review by jlau-neonone — APPROVED** [review]
This worked for me.

**Comment by Viserion77** [comment]
I joined `node_modules` and edit directly the `dynamodb-localhost` files with the changes of this PR and it worked!

**Comment by DeeWBee** [comment]
For those of you directly editing `node_modules` - please understand the next time you run `npm install` or clear your cache (or any equivalent commands), you will overwrite your edits and **it will be broken again**. Please consider using different solutions.

**Comment by sachet-mittal** [comment]
@DeeWBee  I think the solution you are suggesting is a very complicated workaround and adding a new infra (docker) in project which may not be possible for all the projects. The "correct" fix is to patch the `node_modules` using `https://github.com/ds300/patch-package`.  

note: I say this is correct as this would work on local/docker/CI-CD environments with minimal changes but YMMV.  If docker works for you then please continue doing that as you already tackled the most difficult part and made your project more robust.

**Review by AshanFernando — APPROVED** [review]

**Comment by gpmcadam** [comment]
thanks for merging. any idea when this will be published to npm?

**Comment by McKean** [comment]
if anyone needs this to work in your containers (for testing) until the package is updated, just throw those lines into your Dockerfile after installing your packages (`npm install` or `yarn`)
```
# until package is updated
RUN sed -i 's|http://s3-us-west-2.amazonaws.com/dynamodb-local/dynamodb_local_latest.tar.gz|https://s3.us-west-2.amazonaws.com/dynamodb-local/dynamodb_local_latest.tar.gz|g' node_modules/dynamodb-localhost/dynamodb/config.json
RUN sed -i 's|http|https|g' node_modules/dynamodb-localhost/dynamodb/installer.js
```

**Comment by jonnixt17** [comment]
Or `npm install --save 99x/dynamodb-localhost#db30898f8c40932c7177be7b2f1a81360d12876d` should work while we wait. (Do keep in mind there was at least one additional merge done between the last npm release and the current master.)

**Comment by LeoFalco** [comment]
Helllo, @AshanFernando

Can you publish this fix on npm too ?

**Comment by LeoFalco** [comment]
> Helllo, @AshanFernando
> 
> Can you publish this fix on npm too ?

I wish a new released version to bump dependency of `dynamodb-localhost` lib on package `serverless-dynamodb-local`

**Comment by codebravotech** [comment]
Please merge!

**Comment by voccer** [comment]
the PR was merged, but it is still an old version in the npm package.

**Comment by colesiegel** [comment]
This still doesn't work - was it released?

@AshanFernando

**Comment by tspangler** [comment]
looks like this needs a new npm release

**Comment by domdomegg** [comment]
FYI: Despite this PR being merged, this fix to the package (`dynamodb-localhost`) still hasn't been published, with no new releases to NPM in over 4 years.

**Solution:** You can use [aws-dynamodb-local](https://www.npmjs.com/package/aws-dynamodb-local), a maintained fork, instead. (Disclaimer: I am a contributor to this fork). It is a drop-in replacement for this package, and is updated to fix this bug. If you're using `serverless-dynamodb-local`, you can also replace that with it's equivalent [serverless-dynamodb](https://github.com/raisenational/serverless-dynamodb#migrating-from-serverless-dynamodb-local).

Migrating takes about 2 minutes, with [a full guide in the README](https://github.com/raisenational/aws-dynamodb-local#migrating-from-dynamodb-localhost). Of course, it's all still [open-source](https://github.com/raisenational/aws-dynamodb-local) and MIT licensed. Ownership of this new package sits with a registered charity, that is committed to maintaining the package into the future and is open to contributions from the community.

File: dynamodb/installer.js (hunk 2 above)
Comment (JonatanPotrikus) [reply dynamodb/installer.js:15]: Good idea @huseyin-caglayan-cko

**Pull request merged**